rust/persia-embedding-config/src/lib.rs

```diff
@@ -117,22 +117,37 @@ static PERSIA_INSTANCE_INFO: OnceCell<Arc<InstanceInfo>> = OnceCell::new();
 
 static PERSIA_EMBEDDING_CONFIG: OnceCell<Arc<EmbeddingConfig>> = OnceCell::new();
 
+/// get local ip address by specific socket name or
+/// request the 8.8.8.8:80 to retrieve the ip address from response.
 fn get_local_ip() -> String {
-    let if_addrs = get_if_addrs::get_if_addrs().expect("failed to get local ip address");
-
-    let persia_socket_name = std::env::var("PERSIA_SOCKET_NAME").unwrap_or(String::from("eth0"));
-
-    if let Some(ip_addr) = if_addrs
-        .iter()
-        .find(|x| x.name == persia_socket_name.as_str())
-    {
-        let ip_addr = ip_addr.ip().to_string();
-        return ip_addr;
-    } else {
-        panic!(
-            "failed to get ip of socket name {}, please try other by setting PERSIA_SOCKET_NAME",
-            persia_socket_name
-        );
+    let socket_ifname = std::env::var("SOCKET_IFNAME").unwrap_or(String::from("eth0"));
```

Comment:
it's fine to call it PERSIA_SOCKET_IFNAME

Reply:
Already update